src/main/java/org/jenkinsci/plugins/DependencyCheck/AbstractDependencyCheckBuilder.java

```diff
@@ -86,7 +87,7 @@ public void perform(@Nonnull final Run<?, ?> build,
         } else {
             success = launcher.getChannel().call(new DependencyCheckExecutor(options, listener));
         }
-        if (success) {
+        if (success || preserveBuildSuccessOnScanFailure) {
             build.setResult(Result.SUCCESS);
         } else {
```

Comment:
Is this the right thing to do? Assuming the result was `Result.UNSTABLE` before, this changes the result to `Result.SUCCESS`. This is not what I would expect when enabling an option that is called `... will not be affect build result.`.